ci/test-sriov-secondary-network-aws.sh

```diff
@@ -0,0 +1,554 @@
+    shift 2
+    ;;
+    --aws-service-user)
+    AWS_SERVICE_USER_NAME="$2"
+    shift 2
+    ;;
+    --aws-region)
+    REGION="$2"
+    shift 2
+    ;;
+    --k8s-version)
+    K8S_VERSION="$2"
+    shift 2
+    ;;
+    --setup-only)
+    RUN_SETUP_ONLY=true
+    RUN_ALL=false
+    shift
+    ;;
+    --cleanup-only)
+    RUN_CLEANUP_ONLY=true
+    RUN_ALL=false
+    shift
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+mkdir -p ~/.aws
+cat > ~/.aws/config <<EOF
+[default]
+region = $REGION
+role_arn = $AWS_SERVICE_USER_ROLE_ARN
+source_profile = $AWS_SERVICE_USER_NAME
+output = json
+EOF
+cat > ~/.aws/credentials <<EOF
+[$AWS_SERVICE_USER_NAME]
+aws_access_key_id = $AWS_ACCESS_KEY
+aws_secret_access_key = $AWS_SECRET_KEY
+EOF
+
+THIS_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+ANTREA_CHART="$THIS_DIR/../build/charts/antrea"
+DOCKER_IMAGE_PATH="$THIS_DIR/../antrea-ubuntu.tar"
+SRIOV_SECONDARY_NETWORKS_YAML="$THIS_DIR/../test/e2e-secondary-network/infra/sriov-secondary-networks.yml"
+IP_POOL_YAML="pool1.yaml"
+NAD_YAML="$THIS_DIR/../test/e2e-secondary-network/infra/sriov-network-attachment-definition.yml"
+
+CONTROLPLANE_IP=""
+WORKER_IP=""
+
+CONTROLPLANE_INSTANCE_ID="${CONTROLPLANE_INSTANCE_ID:-}"
+WORKER_INSTANCE_ID="${WORKER_INSTANCE_ID:-}"
+
+CONTROLPLANE_NODE_ENI="${CONTROLPLANE_NODE_ENI:-}"
+WORKER_NODE_ENI="${WORKER_NODE_ENI:-}"
+
+# Function to launch EC2 instance
+launch_ec2_instance() {
+    local instance_name=$1
+    instance_id=$(aws ec2 run-instances \
+        --image-id $AMI_ID \
+        --count 1 \
+        --instance-type $INSTANCE_TYPE \
+        --key-name "$AWS_EC2_SSH_KEY_NAME" \
+        --security-group-ids "$AWS_SECURITY_GROUP" \
+        --subnet-id "$AWS_SUBNET_ID" \
+        --block-device-mappings DeviceName=/dev/sda1,Ebs={VolumeSize=20} \
+        --query "Instances[0].InstanceId" \
+        --output text)
+    echo "$instance_id"
+}
+
+attach_network_interface() {
+    local instance_id=$1
+    local node_type=$2
+    ENI_ID=$(aws ec2 create-network-interface \
+      --subnet-id "$AWS_SUBNET_ID" \
+      --groups "$AWS_SECURITY_GROUP" \
+      --query 'NetworkInterface.NetworkInterfaceId' \
+      --output text)
+
+    echo "Network interface created successfully with ENI ID: $ENI_ID"
+
+    #  Attach the ENI to the EC2 instance
+    echo "Attaching network interface $ENI_ID to instance $instance_id ..."
+    ATTACHMENT_ID=$(aws ec2 attach-network-interface \
+      --network-interface-id "$ENI_ID" \
+      --instance-id "$instance_id" \
+      --device-index 1 \
+      --query 'AttachmentId' \
+      --output text)
+
+    echo "Network interface attached successfully with Attachment ID: $ATTACHMENT_ID"
+
+    if [[ "$node_type" == "control-plane" ]]; then
+        CONTROLPLANE_NODE_ENI="$ENI_ID"
+        echo "Assigned ENI ID $ENI_ID to CONTROLPLANE_NODE_ENI."
```

Comment:
It's unnecessary to append a dot at the end of a log, please check other occurrences and remove them.
```suggestion
        echo "Assigned ENI ID $ENI_ID to CONTROLPLANE_NODE_ENI"
```